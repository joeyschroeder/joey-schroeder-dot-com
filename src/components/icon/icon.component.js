import '@fortawesome/fontawesome-free/css/all.css';

import PropTypes from 'prop-types';
import React from 'react';

export const Icon = (props) => {
  const { className } = props;
  return <i className={className} aria-hidden="true" />;
};

Icon.propTypes = {
  className: PropTypes.string,
};

Icon.defaultProps = {
  className: 'fas fa-file-pdf',
};
