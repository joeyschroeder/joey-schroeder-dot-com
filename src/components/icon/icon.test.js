import React from 'react';
import { create } from 'react-test-renderer';
import { Icon } from './icon.component';

describe('<Icon />', () => {
  it('should render correctly', () => {
    const component = create(<Icon />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
