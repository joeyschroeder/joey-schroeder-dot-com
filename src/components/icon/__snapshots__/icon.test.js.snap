// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render correctly 1`] = `
<i
  aria-hidden="true"
  className="fas fa-file-pdf"
/>
`;
